src/Momento.Sdk/Config/Retry/FixedCountRetryStrategy.cs

```diff
@@ -26,15 +35,23 @@ IRetryStrategy IRetryStrategy.WithLoggerFactory(ILoggerFactory loggerFactory)
 
     public FixedCountRetryStrategy WithMaxAttempts(int maxAttempts)
     {
-        return new(maxAttempts, LoggerFactory);
+        return new(maxAttempts, _eligibilityStrategy, LoggerFactory);
     }
 
-    public int? DetermineWhenToRetryRequest(IGrpcResponse grpcResponse, IGrpcRequest grpcRequest, int attemptNumber)
+    public int? DetermineWhenToRetryRequest<TRequest>(Status grpcStatus, TRequest grpcRequest, int attemptNumber) where TRequest : class
     {
+        _logger.LogDebug($"Determining whether request is eligible for retry; status code: {grpcStatus.StatusCode}, request type: {grpcRequest.GetType()}, attemptNumber: {attemptNumber}, maxAttempts: {MaxAttempts}");
```

Comment:
Here and elsewhere I notice you're charging the users' runtimes string interpolation allocations unconditionally. I think you should rewrite all these LogDebugs like:
```suggestion
        _logger.LogDebug(
            "Determining whether request is eligible for retry; status code: {StatusCode}, request type: {RequestType}, attemptNumber: {AttemptNumber}, maxAttempts: {MaxAttempts}",
            grpcStatus.StatusCode,
            grpcRequest.GetType(),
            attemptNumber,
            MaxAttempts
        );
```

see for more examples and  detail: https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.logdebug?view=dotnet-plat-ext-6.0

Reply:
good call, agreed, it was my intent to write them that way but I always forget.  thx

Reply:
kotlin-structured-logging is an addictive library. It's so nice to not have to care about anything w.r.t. logging invocation cost inside of those logging blocks. I catch myself writing logs like this too so I look out for them

Reply:
i hope we will have a rust-structured-logging (or maybe one already exists in the rust universe)